usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -5546,6 +6249,25 @@ CK_RV ep11tok_sign_init(STDLL_TokData_t * tokdata, SESSION * session,
         goto done;
     }
 
+    rc = ep11tok_pkey_check(tokdata, session, key_obj, mech);
+    switch (rc) {
+    case CKR_OK:
+        rc = sign_mgr_init(tokdata, session, ctx, mech, recover_mode, key);
+        if (rc == CKR_OK)
+            ctx->pkey_active = TRUE;
+        /* Regardless of the rc goto done here, because ep11tok_pkey_check
+         * could have unlocked/relocked the obj so that the blob is no more
+         * valid.
+         */
+        free(ep11_sign_state);
+        goto done;
+    case CKR_FUNCTION_NOT_SUPPORTED:
+        /* ep11 fallback */
+        break;
+    default:
+        goto done;
```

Comment:
Also `free(ep11_sign_state);` here in the default case.

Same allies to the other xxx_init functions with similar switch statement.